### PR TITLE
cleanup adding interpreter paths and make it configurable

### PR DESCRIPTION
**Pull request by mwrock**
This PR accomplishes four things:

1. Fixes a bug where the supervisor will use the most recent version of powershell or busybox-static installed instead of the one packaged. This is fixed with [this line](https://github.com/habitat-sh/habitat/pull/7601/files#diff-9e4f1f1544b6c3ae7b8a78d289f54773R68). My guess is that we used to store the `hab-sup` binary in the root of the package and not in `bin`.
1. Fixes [another bug](https://github.com/habitat-sh/habitat/issues/6410) where the interpreter is not propperly rooted in a local windows studio.
1. No longer allow an interpreter binary that is found on the path but not in a package to be used. I do not think this is a breaking change and do not know why we would allow this.
1. Allows a user to configure the interpreter ident via a `HAB_INTERPRETER_IDENT` env variable or config setting.
Signed-off-by: mwrock <matt@mattwrock.com>

### CODE REVIEW

File: components/common/src/util/path.rs [hunk 1]
```diff
@@ -92,58 +87,47 @@ async fn interpreter_paths() -> Result<Vec<PathBuf>> {
...
+                                                VERSION,
+                                                FS_ROOT_PATH.as_path(),
+                                                &cache_artifact_path(None::<String>),
+                                                None,
+                                                &InstallMode::default(),
+                                                &LocalPackageUsage::default(),
+                                                InstallHookMode::default()).await.is_err() {
+                                                    return Err(Error::InterpreterNotFound(ident));
+                                                }
+                    let pkg_install = PackageInstall::load(&ident, Some(FS_ROOT_PATH.as_ref()))?;
```
Comment (davidMcneil) [inline components/common/src/util/path.rs:113]: Can we use the `PackageInstall` returned from `install::typed_erased_start` or is it important to try and `load` the package?

File: components/common/src/util/path.rs [hunk 2]
```diff
@@ -92,58 +87,47 @@ async fn interpreter_paths() -> Result<Vec<PathBuf>> {
...
+                                                                                .into(),
+                                                &*PROGRAM_NAME,
+                                                VERSION,
+                                                FS_ROOT_PATH.as_path(),
+                                                &cache_artifact_path(None::<String>),
+                                                None,
+                                                &InstallMode::default(),
+                                                &LocalPackageUsage::default(),
+                                                InstallHookMode::default()).await.is_err() {
+                                                    return Err(Error::InterpreterNotFound(ident));
```
Comment (davidMcneil) [inline components/common/src/util/path.rs:111]: We end up losing the underlying reason _why_ the interpreter could not be found. Maybe this is not a big deal. I have never seen it done before but I wonder if we should do something like `InterpreterNotFound(PackageIdent, Box<Self>)`?

Comment (mwrock) [reply components/common/src/util/path.rs]: good call!

File: components/common/src/util/path.rs (hunk 1 above)
Comment (mwrock) [reply components/common/src/util/path.rs]: No that is a great point and I just did not notice that the previous code was not taking advantage of that

**Comment by mwrock** [comment]
I implemented both of your suggestions @davidMcneil

**Pull request merged**